symengine/printers/latex.cpp

```diff
@@ -230,6 +239,61 @@ void LatexPrinter::bvisit(const StrictLessThan &x)
     str_ = s.str();
 }
 
+std::string LatexPrinter::apply2(const DenseMatrix &b)
+{
+    this->bvisit(b);
+    return str_;
+}
+
+void LatexPrinter::bvisit(const DenseMatrix &m) {
+    const int MAX_NUMBER_OF_ROWS = 24;
+    const int MAX_NUMBER_OF_COLUMNS = 16;
+    const int nrows = m.nrows();
+    const int ncols = m.ncols();
+    int nrows_display = nrows;
+    if (nrows > MAX_NUMBER_OF_ROWS) 
+        nrows_display = MAX_NUMBER_OF_ROWS - 2;
+    int ncols_display = MAX_NUMBER_OF_COLUMNS ;
+    if (ncols <ncols_display) 
+        ncols_display = ncols;
+
+    std::ostringstream s;
+    s << "\\left[\\begin{matrix}" << std::endl;
+
+    std::string end_of_line = " \\\\\n";
+    if (ncols_display<ncols) {
+        end_of_line = " & \\cdots "  + end_of_line;
+    }
+    for(int row_index = 0; row_index < nrows_display; row_index++) 
+    {       
+        for(int column_index = 0; column_index < ncols_display; column_index++) 
+        {       
+        RCP< const Basic> v = m.get(row_index, column_index);
+
+        if (v.is_null() ) 
+        {
+            // how to handle this? can happen if DenseMatrix has not been initialized. for now choose to print a question mark
+            s << "?";
```

Comment:
Either that, or you can throw an exception.

Reply:
I think we should throw an exception. We don't handle uninitialized elements like this in other places.

Reply:
@isuruf The initialized elements are indeed not handled properly. The following generates a segmentation fault on my system
```
#include <iostream>
#include <symengine/printers/latex.h>
#include <symengine/printers.h>
#include <symengine/basic.h>
#include <symengine/matrix.h>

using namespace SymEngine;

int main() {
    
    printf("------------------\n");
    DenseMatrix d = DenseMatrix(63,2);
    RCP<const Basic> e = d.get(1,1);
    latex(*e);

    return 0;
}
```
The error is somewhere in `std::string StrPrinter::apply(const Basic &b)`, but it is not directly clear to me where we can add a check to throw a proper exception.